### PR TITLE
Update the TCK generation , remove old shell script (tckbundle.sh)

### PR DESCRIPTION
**Pull request by alwin-joseph**
**Describe the changes:**

- Moved jaxrs-tck-docs as jaxrs-tck/docs
- To generate userguide : `mvn -f jaxrs-tck/docs/userguide` 
- To build TCK jar and zip bundle (with EFTL) : 
`cd jaxrs-tck; mvn clean install`
- To build TCK jar and zip bundle (with EPL) : 
`cd jaxrs-tck; mvn clean install -PEPL`

### CODE REVIEW

**Review by spericas — DISMISSED** [review]
LGTM

File: jaxrs-tck/docs/userguide/src/main/jbake/content/config.inc [hunk 1]
```diff
@@ -0,0 +1,340 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2020, 2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (jim-krueger) [inline jaxrs-tck/docs/userguide/src/main/jbake/content/config.inc:3]: , 2024

File: jaxrs-tck/docs/userguide/src/main/jbake/content/rebuild.inc [hunk 2]
```diff
@@ -0,0 +1,163 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2020, 2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (jim-krueger) [inline jaxrs-tck/docs/userguide/src/main/jbake/content/rebuild.inc:3]: , 2024

File: jaxrs-tck/docs/userguide/src/main/jbake/content/attributes.conf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// :subset-allowed:
+//
+// The environment variable used to specify the home directory
+// for the technology.  Used in config.inc.
+:TechnologyHomeEnv: JAXRS_HOME
+// Java SE version required.
+:SEversion: 11
+:MavenVersion: 3.6.3+
+:JakartaEEVersion: 10
+:excludeListFileName: jaxrs-tck-docs/TCK-Exclude-List.txt
```
Comment (jim-krueger) [inline jaxrs-tck/docs/userguide/src/main/jbake/content/attributes.conf:28]: jaxrs-tck/docs/TCK-Exclude-List.txt

**Review by jim-krueger — DISMISSED** [review]
Needs a PR for main branch as well.

**Review by jansupol — DISMISSED** [review]

**Comment by alwin-joseph** [comment]
> Needs a PR for main branch as well.

Will it be better to re-target this PR to main branch. 
I can create another PR for main branch if that is more convenient. In this case, please help merge this PR.

**Comment by spericas** [comment]
@alwin-joseph I think we should just re-target it to main

**Review by jim-krueger — APPROVED** [review]

**Review by jamezp — COMMENTED** [review]
I don't think this question should block the PR and maybe it's been brought up before, but is there any reason we don't simply just generate the TCK docs for each build instead of keeping the generate static content in the source repository?

File: jaxrs-tck/pom.xml [hunk 4]
```diff
@@ -106,25 +106,70 @@
...
+                <license>EFTL</license>
+            </properties>
+        </profile>
+        <profile>
+            <id>EPL</id>
+            <properties>
+                <bundle-name>restful-ws-tck</bundle-name>
+                <license>EPL</license>
+            </properties>
+        </profile>
```
[the remark this answers is not in the document]
Comment (alwin-joseph) [reply jaxrs-tck/pom.xml:172]: We create Eclipse Public Licenced bundles also for all TCKs other than EFTL bundles.
See section "Materials for a TCK Release" in https://jakarta.ee/committees/specification/tckprocess/

   _Projects MUST produce the project licensed (EPL, Apache, etc.) version for distribution to Maven Central or other open source channels, usable for automated, non-official testing, and implementation. No compatibility claims can be made on the basis of this TCK._

   _Projects MUST produce a final candidate binary that includes the [EFTL](https://www.eclipse.org/legal/tck.php) license. The Jakarta EE Specification Committee will sign and promote project TCK binary for distribution via Eclipse infrastructure on final approval. This is the TCK binary usable for self-certification when one desires to make a claim of compatibility, allowing for the use of the Jakarta brands._

**Comment by alwin-joseph** [comment]
> I don't think this question should block the PR and maybe it's been brought up before, but is there any reason we don't simply just generate the TCK docs for each build instead of keeping the generate static content in the source repository?

If you mean the generation of jaxrs-tck/docs/userguide , IMO keeping it separate can reduce the overall build time of other modules(spec, api,tck, examples). 

Maybe this can be discussed further if the current build process can be improved.

File: jaxrs-tck/pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jamezp) [reply jaxrs-tck/pom.xml:172]: Thank you! I had a feeling it was something like this, but I wasn't sure.

**Comment by jamezp** [comment]
> > I don't think this question should block the PR and maybe it's been brought up before, but is there any reason we don't simply just generate the TCK docs for each build instead of keeping the generate static content in the source repository?
> 
> If you mean the generation of jaxrs-tck/docs/userguide , IMO keeping it separate can reduce the overall build time of other modules(spec, api,tck, examples).
> 
> Maybe this can be discussed further if the current build process can be improved.

FWIW when I built it locally it didn't take much time at all. I would love to discuss that build process as a whole really as well :)

**Review by spericas — APPROVED** [review]

**Pull request merged**